WordLadder/src/assignment3/Main.java

```diff
@@ -139,95 +169,87 @@ public static void initialize() {
 	}
 
 	public static void printLadder(ArrayList<String> ladder) {
-		
-		String start = ladder.get(0);
-		String end = ladder.get(1);
-		int n = ladder.size() - 2;
-		
-		if(ladder.isEmpty()){
-		
-			System.out.println("no word ladder can be found between " 
-			+ start + " and " + end + ".");
+
+		if (ladder.isEmpty()) {
+
+			System.out.println("no word ladder can be found between " + start + " and " + end + ".");
 		}
-		
-		else { 
-			
-			System.out.println("a " + n + "-rung word ladder exists between "
-					+ start + " and " + end + ".");
-		
-			for (int i = 0; i < ladder.size(); i++){
+
+		else {
+
+			System.out.println(
+					"a " + (ladder.size() - 2) + "-rung word ladder exists between " + start + " and " + end + ".");
+
+			for (int i = 0; i < ladder.size(); i++) {
 				System.out.println(ladder.get(i) + "\n");
 			}
 		}
 	}
-	// TODO
-	// Other private static methods here
-	
-	private static boolean recursDFS(Set<String> dict, Set<String> hs,	
-			ArrayList<String> wordL, String start, String end) {
+
+	private static boolean recursDFS(Set<String> dict, Set<String> hs, ArrayList<String> wordL, String start,
+			String end) {
 
 		boolean flag = false;
-		
+
 		int count = 0;
-		int i,j,k = 0;
 		int curr = 0;
-		
+
 		String temp = start;
 		String[] startTemp = start.split("");
 		String[] endTemp = end.split("");
-		
-		if (start.equals(end)){
+
+		if (start.equals(end)) {
 			return true;
-		}		
-		
-		for (i = 0; i < end.length(); i++){
-			
-			if ((startTemp[i]).equals(endTemp[i])){
+		}
+
+		for (int i = 0; i < end.length(); i++) {
+
+			if ((startTemp[i]).equals(endTemp[i])) {
 				curr = i + 1;
-				
-				if (curr >= end.length()){
+
+				if (curr >= end.length()) {
 					curr = 0;
 					break;
 				}
 			}
 		}
-		
-		while (count != start.length() - 1){
-			
-			for (j = 'a'; j < 'z'; j++){
+
+		while (count != start.length() - 1) {
+
+			for (int j = 'A'; j < 'Z'; j++) {
```

Comment:
Looks good, but I think I previously made the mistake of making j < 'Z', instead of j<= 'Z', correct me if I am wrong, thanks.